pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -240,6 +241,17 @@ private TableDataManager createTableDataManager(String tableNameWithType, TableC
   @Override
   public void deleteTable(String tableNameWithType)
       throws Exception {
+
+    _tableDataManagerMap.compute(tableNameWithType, (k, v) -> {
```

Comment:
(MAJOR) This will break. We cannot use `compute()` here because it will remove the entry from the map. Use `get()` instead